pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -276,6 +279,12 @@
     private Set<BrokerEntryMetadataInterceptor> brokerEntryMetadataInterceptors;
     private Set<ManagedLedgerPayloadProcessor> brokerEntryPayloadProcessors;
 
+    private final CopyOnWriteArrayList<Consumer<PublishBufferEvent>>
```

Comment:
I would prefer a better data structure than `CopyOnWriteArrayList` if we have a lot of producers.